src/Encoder/NetscapeBookmarkEncoder.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Shaarli Netscape Bookmark Parser package.
+ *
+ * (c) "Matthias Morin" <mat@tangoman.io>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Shaarli\NetscapeBookmarkParser\Encoder;
+
+/**
+ * This encoder will export array to a flat Netscape-bookmark-file.
+ *
+ * @author "Matthias Morin" <mat@tangoman.io>
+ */
+class NetscapeBookmarkEncoder
+{
+    public const FORMAT = 'netscape';
+
+    // phpcs:disable Generic.Files.LineLength
+    public const HEADER = "<!DOCTYPE NETSCAPE-Bookmark-file-1>\n<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=UTF-8\">\n<TITLE>Bookmarks</TITLE>\n<H1>Bookmarks</H1>\n<DL><p>\n";
+    // phpcs:enable
+
+    /**
+     * Checks whether the serializer can encode to given format.
+     *
+     * @param string $format Format name
+     *
+     * @return bool
+     */
+    public function supportsEncoding(string $format): bool
+    {
+        return self::FORMAT === $format;
+    }
+
+    /**
+     * Encodes data into the given format.
+     *
+     * @param mixed $data    Data to encode
+     * @param array $context Options that normalizers/encoders have access to
+     *
+     * @return string
+     */
+    public function encode($data, array $context = []): string
+    {
+        if (!\is_array($data)) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'Argument 1 passed to %s::%s must be of type array, %s given',
+                    self::class,
+                    __FUNCTION__,
+                    \gettype($data)
+                )
+            );
+        }
```

Comment:
Why don't you type `$data` as an array instead?

Reply:
Well, I could but I want to have this class compatible with Symfony encoder interface in the future... Hmmm... Should I forget about it ?

Reply:
Not that fine, didn't think of that.